js/customize-object-selector-component.js

```diff
@@ -111,6 +111,7 @@ wp.customize.ObjectSelectorComponent = (function( api, $ ) {
 						return $.trim( component.select2_result_template( data ) );
 					},
 					templateSelection: function( data ) {
+						data.multiple = component.select2_options.multiple;
```

Comment:
Is there a better way how to get it from template that it's multiple select field?